yang/vyatta-protocols-frr-link-params-v1.yang

```diff
@@ -0,0 +1,229 @@
+				must "../max-bandwidth" {
+					error-message "Maximum Bandwidth must be set";
+				}
+				must "../available-bandwidth <= ../max-bandwidth" {
+					error-message "Available Bandwidth cannot be greater than Maximum Bandwidth";
+				}
+			}
+			leaf delay {
+				type uint32 {
+					range "0..16777215";
+				}
+				units "micro-second";
+				configd:help "Unidirectional Average Link Delay (micro-second)";
+				description "Unidirectional Average Link Delay";
+			}
+			leaf delay-variation {
+				type uint32 {
+					range "0..16777215";
+				}
+				units "micro-second";
+				configd:help "Unidirectional Link Delay Variation (micro-second)";
+				description "Unidirectional Link Delay Variation";
+			}
+			leaf max-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Maximum bandwidth that can be used (Bytes/second)";
+				description "Maximum bandwidth that can be used";
+			}
+			leaf max-reserved-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Maximum bandwidth that may be reserved (Bytes/second)";
+				description "Maximum bandwidth that may be reserved. It is the bandwidth that can
+					theoretically be reserved";
+				must "../max-bandwidth" {
+					error-message "Maximum Bandwidth must be set";
+				}
+				must "../max-reserved-bandwidth <= ../max-bandwidth" {
+					error-message "Maximum Reservable Bandwidth cannot be greater than Maximum Bandwidth";
+				}
+			}
+			leaf mpls-te-metric {
+				type uint32 {
+					range "1..max";
+				}
+				configd:help "Link metric for MPLS-TE purpose";
+				description "Link metric for MPLS-TE purpose";
+			}
+			leaf neighbor {
+				type types:ipv4-address;
+				configd:help "Configure remote ASBR information (Neighbor IP address and AS number)";
+				description "Configure remote ASBR information (Neighbor IP address and AS number)";
+			}
+			leaf packet-loss {
+				type uint32 {
+					range 0..100;
+				}
+				units "Percentage";
+				configd:help "Unidirectional Link Packet Loss (Percentage)";
+				description "Unidirectional Link Packet Loss (Percentage)";
+			}
+			leaf residual-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Unidirectional Residual Bandwidth (Bytes/second)";
+				description "Unidirectional Residual Bandwidth. For a link or forwarding adjacency, residual
+					bandwidth is defined to be Maximum Bandwidth [RFC3630] minus the bandwidth currently allocated
+					to RSVP-TE LSPs. For a bundled link, residual bandwidth is defined to be the sum of the
+					component link residual bandwidths [draft-ietf-isis-te-metrics-extension-03.txt].";
+				must "../max-bandwidth" {
+					error-message "Maximum Bandwidth must be set";
+				}
+				must "../residual-bandwidth <= ../max-bandwidth" {
+					error-message "Residual Bandwidth cannot be greater than Maximum Bandwidth";
+				}
+			}
+			container unreserved-bandwidth {
```

Comment:
If max-bandwidth is X, it seems you can have X unreserved bandwidth at each of the 8 priority levels, making 8X?

Reply:
![image](https://user-images.githubusercontent.com/79829103/119960019-29284e00-bfc2-11eb-8235-52f72ab8fd57.png)
In each of the priority levels, the bandwidth can be set equal to max-bandwidth. The same way has been implemented in FRR.

Reply:
That's as maybe. I'd like you to explain to me why this is correct, rather than just saying it's done this way somewhere else.

Reply:
The [RFC3630] has been followed here wherein section 2.5.8 it stated about Unreserved Bandwidth as follows:

The Unreserved Bandwidth sub-TLV specifies the amount of bandwidth not yet reserved at each of the eight priority levels in IEEE floating-point format.  The values correspond to the bandwidth that can be reserved with a setup priority of 0 through 7, arranged in increasing order with priority 0 occurring at the start of the sub-TLV, and priority 7 at the end of the sub-TLV.  The initial values (before any bandwidth is reserved) are all set to the Maximum Reservable Bandwidth.  Each value will be less than or equal to the Maximum Reservable Bandwidth.  The units are bytes per second.
The Unreserved Bandwidth sub-TLV is TLV type 8, and is 32 octets in length.

Reply:
I'm still not convinced this is config as opposed to state information.

Reply:
or at least information that should be dynamically derived not set in stone in the config